tools/generate_from_protos.sh

```diff
@@ -70,4 +67,52 @@ for plugin in ${plugin_list}; do
     ${protoc_binary} -I ${proto_dir} --custom_out=${tmp_output_dir} --plugin=protoc-gen-custom=${protoc_gen_dcsdk} --custom_opt="file_ext=h,template_path=${template_path_mavsdk_server}" ${proto_dir}/${plugin}/${plugin}.proto
     mkdir -p ${script_dir}/../src/backend/src/plugins/${plugin}
     mv ${tmp_output_dir}/${plugin}/$(snake_case_to_camel_case ${plugin}).h ${script_dir}/../src/backend/src/plugins/${plugin}/${plugin}_service_impl.h
+
+    file_impl_h="${script_dir}/../src/plugins/${plugin}/${plugin}_impl.h"
+    if [[ ! -f "${file_impl_h}" ]]; then
+        ${protoc_binary} -I ${proto_dir} --custom_out=${tmp_output_dir} --plugin=protoc-gen-custom=${protoc_gen_dcsdk} --custom_opt="file_ext=h,template_path=${template_path_plugin_impl_h}" ${proto_dir}/${plugin}/${plugin}.proto
+        mv ${tmp_output_dir}/${plugin}/$(snake_case_to_camel_case ${plugin}).h ${file_impl_h}
+        echo "-> Creating ${file_impl_h}"
+    else
+        # Warn if file is not checked in yet.
+        if [[ ! $(git ls-files --error-unmatch ${file_impl_h} 2> /dev/null) ]]; then
+            echo "-> Not creating ${file_impl_h} because it already exists"
+        fi
+    fi
+
+    file_impl_cpp="${script_dir}/../src/plugins/${plugin}/${plugin}_impl.cpp"
+    if [[ ! -f $file_impl_cpp ]]; then
+        ${protoc_binary} -I ${proto_dir} --custom_out=${tmp_output_dir} --plugin=protoc-gen-custom=${protoc_gen_dcsdk} --custom_opt="file_ext=cpp,template_path=${template_path_plugin_impl_cpp}" ${proto_dir}/${plugin}/${plugin}.proto
+        mv ${tmp_output_dir}/${plugin}/$(snake_case_to_camel_case ${plugin}).cpp ${file_impl_cpp}
+        echo "-> Creating ${file_impl_cpp}"
+    else
+        # Warn if file is not checked in yet.
+        if [[ ! $(git ls-files --error-unmatch ${file_impl_cpp} 2> /dev/null) ]]; then
+            echo "-> Not creating ${file_impl_cpp} because it already exists"
+        fi
+    fi
+
+    file_cmake="${script_dir}/../src/plugins/${plugin}/CMakeLists.txt"
+    if [[ ! -f $file_cmake ]]; then
+        ${protoc_binary} -I ${proto_dir} --custom_out=${tmp_output_dir} --plugin=protoc-gen-custom=${protoc_gen_dcsdk} --custom_opt="file_ext=txt,template_path=${template_path_cmake}" ${proto_dir}/${plugin}/${plugin}.proto
+        mv ${tmp_output_dir}/${plugin}/$(snake_case_to_camel_case ${plugin}).txt ${file_cmake}
+        echo "-> Creating ${file_cmake}"
+    else
+        # Warn if file is not checked in yet.
+        if [[ ! $(git ls-files --error-unmatch ${file_cmake} 2> /dev/null) ]]; then
+            echo "-> Not creating ${file_cmake} because it already exists"
+        fi
+    fi
+
+    plugins_cmake_file="${script_dir}/../src/plugins/CMakeLists.txt"
+    if [[ ! $(grep ${plugin} ${plugins_cmake_file}) ]]; then
+        echo "-> Adding entry for '${plugin}' to ${plugins_cmake_file}"
+
+        # We want to append the plugin to the list but before the passthrough plugin.
```

Comment:
We will lose the alphabetical order :scream: :scream: :scream: 

Just kidding :grin: :trollface:

Reply:
Yer, I thought about how to make that happen but gave up.